lib/p2p/Hosts.ts

```diff
@@ -0,0 +1,19 @@
+/** Represents a list of hosts for managing network peers activity */
+class Hosts {
```

Comment:
I'd written a comment here but evidently I forgot to save it or something.  I think it makes more sense to have this class be of the singular form to be consistent with `Peer`, so instead of `hosts.ban(key)` we'd do `host[key].ban()`.

I think there's also a question of whether we want to be able to ban by host/IP address, node key, or both. But that's somewhat outside the scope of this PR and we can address that later.

Reply:
I see what you're saying, i'll need to think on some other aspects before deciding what's the best way of doing it. I still haven't thought all the way through on that one. Wanted to save it for the next PR.
The design will anyway need to easily support either of the banning options you mentioned. So yes, the question is more about our general strategy.